File: RepoTransBench/target_projects/C/Python/dfu-programmer_dfu-programmer/tests/original/test_util.py
import pytest
import sys
from src import util

def test_dfu_debug_level_high(capfd, monkeypatch):
    """
    Corresponds to C's test_dfu_debug_level_high.
    Tests that a debug message with a level higher than the global debug level is NOT printed.
    """
    monkeypatch.setattr(util, '_debug_level', 100) # Set global debug level
    util.dfu_debug("testfile", "testfunc", 42, 150, "should not print: %d\n", 123)
    captured = capfd.readouterr() # Capture stdout and stderr
    assert captured.out == ""
    assert captured.err == ""

def test_dfu_debug_level_low(capfd, monkeypatch):
    """
    Corresponds to C's test_dfu_debug_level_low.
    Tests that a debug message with a level lower than or equal to the global debug level IS printed.
    """
    monkeypatch.setattr(util, '_debug_level', 100) # Set global debug level
    util.dfu_debug("testfile", "testfunc", 42, 50, "message: %s\n", "low-level debug")
    captured = capfd.readouterr() # Capture stdout and stderr
    assert "[testfile:testfunc:42] message: low-level debug\n" in captured.out
    assert captured.err == ""

File: RepoTransBench/target_projects/C/Rust/dfu-programmer_dfu-programmer/tests/original/test_util.rs
use dfu_programmer::util::{dfu_debug, dfu_debug_int};
use std::sync::Once;

static INIT: Once = Once::new();

// Setup function to initialize any test dependencies
fn initialize() {
    INIT.call_once(|| {
        // Set global debug level
        unsafe {
            dfu_programmer::util::DEBUG = 100;
        }
    });
}

#[test]
fn test_dfu_debug_level_high() {
    initialize();
    // This message should not print, because debug == 100, level=150
    dfu_debug_int("testfile", "testfunc", 42, 150, "should not print", 123);
    // No direct assertion, we're just making sure it doesn't crash
    assert!(true);
}

#[test]
fn test_dfu_debug_level_low() {
    initialize();
    // This message should print, because debug == 100, level=50
    dfu_debug("testfile", "testfunc", 42, 50, "message", "low-level debug");
    // No direct assertion, we're just making sure it doesn't crash
    assert!(true);
}